src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/comparison.cpp

```diff
@@ -0,0 +1,192 @@
+
+namespace ov {
+namespace test {
+std::string ComparisonLayerCPUTest::getTestCaseName(const testing::TestParamInfo<ComparisonLayerCPUTestParamSet> &obj) {
+    const auto& [shapes, comparisonType, secondInType, modelPrc, inferPrc, enforceSnippets] = obj.param;
+
+    std::ostringstream result;
+    result << "IS=(";
+    for (const auto& shape : shapes) {
+        result << ov::test::utils::partialShape2str({shape.first}) << "_";
+    }
+    result << ")_TS=(";
+    for (const auto& shape : shapes) {
+        for (const auto& item : shape.second) {
+            result << ov::test::utils::vec2str(item) << "_";
+        }
+    }
+    result << "OpType=" << comparisonType << "_";
+    result << "secondInType=" << secondInType << "_";
+    result << "modelPrc=" << modelPrc.to_string() << "_";
+    result << "inferPrc=" << inferPrc.to_string() << "_";
+    result << "_enforceSnippets=" << enforceSnippets;
+
+    return result.str();
+}
+
+void ComparisonLayerCPUTest::SetUp() {
+    const auto& [shapes, comparisonType, secondInType, modelPrc, inferPrc, enforceSnippets] = this->GetParam();
+    targetDevice = ov::test::utils::DEVICE_CPU;
+
+    // we have to change model precision as well, otherwise inference precision won't affect single-node graph
+    // due to enforce inference precision optimization for the eltwise as first node of the model
+    const auto currModelPrc = ov::element::Type(modelPrc).is_real() ? inferPrc : modelPrc;
+
+    const auto primitiveType = getPrimitiveType(comparisonType, currModelPrc);
+    const auto primiticePrcType =
+        ov::intel_cpu::any_of(primitiveType, "ref", "acl") ? ov::element::f32.to_string() : currModelPrc.to_string();
+    selectedType = primitiveType.empty() ? "" : primitiveType + "_" + primiticePrcType;
+
+    init_input_shapes(shapes);
+
+    if (enforceSnippets) {
+        configuration.insert(ov::intel_cpu::snippets_mode(ov::intel_cpu::SnippetsMode::IGNORE_CALLBACK));
+    } else {
+        configuration.insert(ov::intel_cpu::snippets_mode(ov::intel_cpu::SnippetsMode::DISABLE));
+    }
+    configuration.insert(ov::hint::inference_precision(inferPrc));
+
+    ov::ParameterVector params {std::make_shared<ov::op::v0::Parameter>(currModelPrc, inputDynamicShapes[0])};
+
+    std::shared_ptr<ov::Node> secondInput;
+    if (ov::test::utils::InputLayerType::CONSTANT == secondInType) {
+        auto tensor = ov::test::utils::create_and_fill_tensor(currModelPrc, targetStaticShapes[0][1]);
+        secondInput = std::make_shared<ov::op::v0::Constant>(tensor);
+    } else {
+        auto param = std::make_shared<ov::op::v0::Parameter>(currModelPrc, inputDynamicShapes[1]);
+        secondInput = param;
+        params.push_back(param);
+    }
+    const auto comparison_node = ov::test::utils::make_comparison(params[0], secondInput, comparisonType);
+
+    comparison_node->get_rt_info() = getCPUInfo();
+    function = std::make_shared<ov::Model>(comparison_node, params, "Comparison");
```

Comment:
NIT. Usually we joint these two actions and call `makeNgraphFunction` base class method.

Reply:
Thank you, done!

P.S. Time has come to rename this helper to `makeOVModel` 😄